api/src/routers/meals.js

```diff
@@ -6,7 +6,66 @@ const mealsRouter = express.Router();
 //return all meals
 mealsRouter.get("/", async (req, res, next) => {
   try {
-    const meals = await knex("Meal");
+    const query = knex("meal");
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir,
+    } = req.query;
+    console.log(`maxPrice = ${maxPrice}`);
+    console.log(`title = ${title}`);
+    console.log(`limit = ${limit}`);
+    console.log(`dateafter = ${dateAfter}`);
+    console.log(`datebefore = ${dateBefore}`);
+    console.log(`availableReservation = ${availableReservations}`);
+
+    if (!isNaN(maxPrice)) {
+      query.where("price", "<", maxPrice);
+    }
+    if (availableReservations !== undefined) {
```

Comment:
Let's replace `!== undefined` with `isNaN` where possible

```suggestion
    if (!isNaN(availableReservations)) {
```